src/components/TodoList/TodoList.tsx

```diff
@@ -1,210 +1,116 @@
 /* eslint-disable max-len */
-import React from 'react';
+import classNames from 'classnames';
+import { useDispatch } from 'react-redux';
+import React, { useEffect, useState } from 'react';
+import { Todo } from '../../types/Todo';
+import { useAppSelector } from '../../app/hooks';
+import { Status } from '../../types/Status';
+import { actions } from '../../features/currentTodo';
 
 export const TodoList: React.FC = () => {
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>();
+
+  const dispatch = useDispatch();
+  const filter = useAppSelector(state => state.filter);
+  const todos = useAppSelector(state => state.todos);
+  const currentTodo = useAppSelector(state => state.currentTodo);
+
+  const filterTodos = () => {
+    return todos.filter((todo) => {
+      const filterQuery = todo.title.toLowerCase()
+        .includes(filter.query.toLowerCase());
+
+      switch (filter.status) {
+        case Status.ACTIVE:
+          return !todo.completed && filterQuery;
+        case Status.COMPLETED:
+          return todo.completed && filterQuery;
+        default:
+          return filterQuery;
+      }
+    });
+  };
```

Comment:
Not fully understand comment. No need to call Array.filter() at all in case status is all